src/output.jl

```diff
@@ -8,6 +8,8 @@ png(plt::Plot, fn::AbstractString) =
 
 png(fn::AbstractString) = png(current(), fn)
 
+png(plt::Plot, io::IOBuffer) = (seekstart(io); show(io, MIME("image/png"), plt); seekstart(io))
```

Comment:
Why are the `seekstart(io)` required ?

I'd rather have something like:
```julia
png(io::IO, plt::Plot) = show(io, MIME("image/png"), plt)
```

Reply:
IOBuffers are strange. Sequential reads following writes and subsequent writes all start at buffer's end. The seekstarts are to make sure the read_from_png() gets the proper data.

I agree we could add for all. I did not know for sure which backends supported what types.  Would a PR to output.jl be able to assume that all types were supported?

Reply:
In order to have Cairo's read_from_png() work with an IO, the png must be the entire contents of the IO, as far as I can see.

This means that if we generalize the methods to work with any IO we are going to clobber the contents of a file handle, if that is the IO the function gets from user. I wonder if we may prevent mistakes if we keep the IO as an IOBuffer. I could rename the `io` term to `buffer` to emphasize this. Or do you think that write image to any stream (IO) is a feature we should want, and users should be able to decide if they want to clobber an existing stream?

Reply:
> IOBuffers are strange. Sequential reads following writes and subsequent writes all start at buffer's end. The seekstarts are to make sure the read_from_png() gets the proper data.

Did you try using a [`PipeBuffer`](https://docs.julialang.org/en/v1/base/io-network/#Base.PipeBuffer) instead ?

Reply:
Hm, a PipeBuffer works without a seek.  So either we could specify a PipeBuffer and not use seekstart, or keep the seeks and use a generic IO (a rw file handle would need the seeks).

Reply:
I'd prefer if we used the generic `io::IO`, without any `seekstart` and it would be up to the user to either use a `PipeBuffer`or use a `IOBuffer` and surround the `png` call with `seekstart` if necessary.

Reply:
Corresponding tests should be added to https://github.com/JuliaPlots/Plots.jl/blob/6087ab960fac0587c9e8ccca3c5a89b639a106ac/test/test_output.jl#L5.